libc-top-half/musl/arch/wasm32/pthread_arch.h

```diff
@@ -1,4 +1,8 @@
+#ifdef _REENTRANT
 extern _Thread_local struct __pthread __wasilibc_pthread_self;
+#else
+extern struct __pthread __wasilibc_pthread_self;
+#endif
```

Comment:
Is this needed?  IIUC, TLS stuff just gets lowered away to normal storage when threads are disabled.